I've pushed an update that gets `nfd` building (and expands the comment on `tev`).

Regarding trying to build `tev`, adopting wegank's decision got this most of the way, though it still eventually runs into the error below. Light searching suggests it's an SDK error. I can't use `apple_sdk_11_0` on this system, so that's where the path ends for me. I've pushed up the WIP in https://github.com/abathur/nixpkgs/commit/f828b65015861578fa8c7f8886b0173496ca2b2d and updated

<details><summary>no matching function error; SDK?</summary>

```console
[ 92%] Building CXX object CMakeFiles/tev.dir/src/FalseColor.cpp.o
/tmp/nix-build-tev-1.23.drv-0/source/src/Common.cpp:59:12: error: no viable conversion from returned value of type 'std::__1::u8string' (aka 'basic_string<char8_t>') to function return type 'fs::path'
    return toU8string(utf8);
           ^~~~~~~~~~~~~~~~
/nix/store/nxr7nki8inql0ab7kqbm5457h6aq0isr-libcxx-11.1.0-dev/include/c++/v1/filesystem:763:29: note: candidate constructor not viable: no known conversion from 'std::__1::u8string' (aka 'basic_string<char8_t>') to 'const std::__1::__fs::filesystem::path &' for 1st argument
  _LIBCPP_INLINE_VISIBILITY path(const path& __p) : __pn_(__p.__pn_) {}
                            ^
/nix/store/nxr7nki8inql0ab7kqbm5457h6aq0isr-libcxx-11.1.0-dev/include/c++/v1/filesystem:764:29: note: candidate constructor not viable: no known conversion from 'std::__1::u8string' (aka 'basic_string<char8_t>') to 'std::__1::__fs::filesystem::path &&' for 1st argument
  _LIBCPP_INLINE_VISIBILITY path(path&& __p) noexcept
                            ^
/nix/store/nxr7nki8inql0ab7kqbm5457h6aq0isr-libcxx-11.1.0-dev/include/c++/v1/filesystem:768:3: note: candidate constructor not viable: no known conversion from 'std::__1::u8string' (aka 'basic_string<char8_t>') to 'std::__1::__fs::filesystem::path::string_type &&' (aka 'basic_string<char> &&') for 1st argument
  path(string_type&& __s, format = format::auto_format) noexcept
  ^
/nix/store/nxr7nki8inql0ab7kqbm5457h6aq0isr-libcxx-11.1.0-dev/include/c++/v1/filesystem:772:3: note: candidate template ignored: requirement '__is_pathable<std::__1::basic_string<char8_t, std::__1::char_traits<char8_t>, std::__1::allocator<char8_t>>, false, false, false>::value' was not satisfied [with _Source = std::__1::basic_string<char8_t, std::__1::char_traits<char8_t>, std::__1::allocator<char8_t>>]
  path(const _Source& __src, format = format::auto_format) {
  ^
/nix/store/nxr7nki8inql0ab7kqbm5457h6aq0isr-libcxx-11.1.0-dev/include/c++/v1/string:875:5: note: candidate function
    operator __self_view() const _NOEXCEPT { return __self_view(data(), size()); }
    ^
/tmp/nix-build-tev-1.23.drv-0/source/src/Common.cpp:66:12: error: no matching function for call to 'fromU8string'
    return fromU8string(path.u8string());
           ^~~~~~~~~~~~
/tmp/nix-build-tev-1.23.drv-0/source/src/Common.cpp:35:8: note: candidate function not viable: no known conversion from 'basic_string<char, char_traits<char>, allocator<char>>' to 'const basic_string<char8_t, char_traits<char8_t>, allocator<char8_t>>' for 1st argument
string fromU8string(const u8string& str) {
       ^
[ 92%] Building CXX object CMakeFiles/tev.dir/src/HelpWindow.cpp.o
2 errors generated.
make[2]: *** [CMakeFiles/tev.dir/build.make:280: CMakeFiles/tev.dir/src/Common.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:558: CMakeFiles/tev.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
error: builder for '/nix/store/d5jaj5br8pna7jsgb41a5h6f5hw072wm-tev-1.23.drv' failed with exit code 2
```
</details>

--- a/pkgs/development/lua-modules/nfd/default.nix
+++ b/pkgs/development/lua-modules/nfd/default.nix
@@ -1,5 +1,5 @@
-{ fetchFromGitHub, buildLuarocksPackage, lua, pkg-config, lib
-, substituteAll, zenity }:
+{ stdenv, fetchFromGitHub, buildLuarocksPackage, lua, pkg-config, lib
+, substituteAll, zenity, AppKit }:
 
 buildLuarocksPackage {
   pname = "nfd";
@@ -24,6 +24,8 @@ buildLuarocksPackage {
 
   extraVariables.LUA_LIBDIR = "${lua}/lib";
   nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = lib.optionals stdenv.isDarwin [ AppKit ];
 
   fixupPhase = ''
     find $out -name nfd_zenity.so -execdir mv {} nfd.so \;

--- a/pkgs/top-level/lua-packages.nix
+++ b/pkgs/top-level/lua-packages.nix
@@ -95,6 +95,7 @@ in
 
   nfd = callPackage ../development/lua-modules/nfd {
     inherit (pkgs.gnome) zenity;
+    inherit (pkgs.darwin.apple_sdk.frameworks) AppKit;
   };
 
   vicious = (callPackage ({ fetchFromGitHub }: stdenv.mkDerivation rec {

